service/service_test.go

```diff
@@ -226,7 +226,7 @@ func TestServiceTelemetryWithOpenTelemetryMetrics(t *testing.T) {
 			registry := featuregate.NewRegistry()
 			obsreportconfig.RegisterInternalMetricFeatureGate(registry)
 			colTel := newColTelemetry(registry)
-			require.NoError(t, colTel.registry.Apply(map[string]bool{obsreportconfig.UseOtelForInternalMetricsfeatureGateID: true}))
+			require.NoError(t, colTel.registry.Apply(map[string]bool{obsreportconfig.UseOtelForInternalMetricsfeatureGateID: false}))
```

Comment:
Why is this changed here?

Reply:
Thanks for the review! Not sure why I left this file like this. I think I noticed `assertMetrics` checks for labels in every metric that doesn't apply when the featuregate is enabled. So this test will fail for otel. Instead of just flipping the featuregate off I am now checking if the featuregate is disabled before calling `assertMetrics`. I am definitely open to suggestions for fixing/improving this test.